src/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs

```diff
@@ -18,7 +18,10 @@
 namespace Microsoft.DotNet.Tools.Tool.Install
 {
     internal delegate IShellShimRepository CreateShellShimRepository(DirectoryPath? nonGlobalLocation = null);
-    internal delegate (IToolPackageStore, IToolPackageInstaller) CreateToolPackageStoreAndInstaller(DirectoryPath? nonGlobalLocation = null);
+
+    internal delegate (IToolPackageStore, IToolPackageInstaller) CreateToolPackageStoreAndInstaller(
+        DirectoryPath? nonGlobalLocation = null,
+        IEnumerable<string> forwardRestoreArgument = null);
```

Comment:
Nit: if we keep this argument name, it should probably be pluralized to match `_forwardRestoreArguments`.

Reply:
fixed, changed to suggested